pandas/io/sql.py

```diff
@@ -1222,11 +1225,16 @@ def to_sql(self, frame, name, if_exists='fail', index=True,
         chunksize : int, default None
             If not None, then rows will be written in batches of this size at a
             time.  If None, all rows will be written at once.
-        dtype : dict of column name to SQL type, default None
+        dtype : single SQL type or dict of column name to SQL type, default None
             Optional specifying the datatype for columns. The SQL type should
-            be a SQLAlchemy type.
+            be a SQLAlchemy type.If all columns are of the same type, one 
+            single value can be used.
+
 
         """
+        if dtype and not is_dictlike(dtype):
+            temp_type = dtype
```

Comment:
this `temp_type` is not needed AFAIK, you can just use `dtype` in the line below